Line/android/LINELoginDemo/app/src/main/java/com/google/firebase/linelogindemo/util/LineLoginHelper.java

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.linelogindemo.util;
+
+import android.app.Activity;
+import android.content.Context;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.google.android.gms.tasks.Continuation;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.TaskCompletionSource;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.UserProfileChangeRequest;
+import com.google.firebase.linelogindemo.R;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+
+import jp.line.android.sdk.LineSdkContext;
+import jp.line.android.sdk.LineSdkContextManager;
+import jp.line.android.sdk.api.ApiClient;
+import jp.line.android.sdk.api.ApiRequestFuture;
+import jp.line.android.sdk.api.ApiRequestFutureListener;
+import jp.line.android.sdk.api.FutureStatus;
+import jp.line.android.sdk.exception.LineSdkLoginException;
+import jp.line.android.sdk.login.LineAuthManager;
+import jp.line.android.sdk.login.LineLoginFuture;
+import jp.line.android.sdk.login.LineLoginFutureListener;
+import jp.line.android.sdk.model.Profile;
+
+public class LineLoginHelper {
+
+    private String mLineAcessscodeVerificationEndpoint;
+    private static final String TAG = LineLoginHelper.class.getCanonicalName();
```

Comment:
nit: TAG can be max 23 characters, use `getSimpleName()` to avoid having the package prefix.